turbinia/evidence.py

```diff
@@ -965,6 +966,32 @@ def __init__(self, plaso_version=None, *args, **kwargs):
     self.copyable = True
     self.plaso_version = plaso_version
 
+  def validate(self):
```

Comment:
This overrides an existing validate method that has some other checks in it that would be good to keep if possible.  Should we rename this to something like `validate_final()` or `validate_post()` or something and call it from the other validate method so we can keep both checks but still allow evidence to have this per-evidence type validation?

Reply:
Actually, now that I think about it, I'm not sure it makes sense to call this new `validate*` method from the old one because the old one runs before the Task runs and the new one runs after, so it won't have access to all the right things.  Maybe we just separate them into two different ones for now.  LMK what you think.

Reply:
There are two ways we can do it, if we only call this new method prior to processing the evidence (this is because evidence_setup) calls velidate(), then tasks for which the input evidence fails validation, will appear as failed tasks since we raise a TurbiniaException upon failure to validate. Alternatively, we can run the new method prior to add_evidence only and catch the exception there, in which case tasks would succeed but the output will be ignored (with a log message) and the failed evidence will not appear in saved_paths attributes.

No approach is cleaner than the other but the first approach will fail the tasks due to the expcetion and it would be more obvious what happened, the error (evidence validation failed) will be displayed in the status and in the UI without further changes. The other approach will not cause failed tasks but i'm not sure how to bubble up the exception message on a successful task at that point.

Reply:
IMO some failed tasks are normal, especially if we clearly state why, and it is a totally valid reason. Also seems cleaner that way from a code change perspective.

Reply:
Just for the record since we talked out of band:  We decided to validate the evidence after the task has run but before the result has been sent back to the server.  That way new tasks won't be created to process invalid or empty evidence objects.